### PR TITLE
[EXPLORER] Add tooltip for show desktop button

### PR DESCRIPTION
**Pull request by TAN-Gaming**
After:
VirtualBox_ReactOS_30_10_2024_12_06_59

JIRA issue: [CORE-15369](https://jira.reactos.org/browse/CORE-15369)

### TODO
- [x] Investigate tooltip on Windows 7

### CODE REVIEW

**Review by whindsaks — APPROVED** [review]
Looks fine to me but we should perhaps use a tool like [MiniSpy](https://www.codeproject.com/articles/1732/minispy) to try to figure out if Windows has one tooltip window for everything on the taskbar or if each feature has its own tip HWND.

**Review by binarymaster — COMMENTED** [review]
Make sure to associate PR with ticket [CORE-15369](https://jira.reactos.org/browse/CORE-15369)

**Comment by HBelusca** [comment]
why the "slash Windows" part?

**Comment by binarymaster** [comment]
I think it implies "restore windows"

**Comment by TAN-Gaming** [comment]
> why the "slash Windows" part?

On Win7, the tooltip is always "Show desktop". I think appending "slash Windows" is better because what it actually does is "Show desktop / Restore windows".

**Comment by TAN-Gaming** [comment]
> Make sure to associate PR with ticket [CORE-15369](https://jira.reactos.org/browse/CORE-15369)

Updated PR description.

**Comment by whindsaks** [comment]
> try to figure out if Windows has one tooltip

On Windows 10 I can't even get it to display a tooltip here (Peek on or off does not matter).

**Comment by TAN-Gaming** [comment]
> On Windows 10 I can't even get it to display a tooltip here (Peek on or off does not matter).

Yes, Win10 doesn't show the tooltip but Win7 shows the tooltip in both classic and aero theme.
Win7ShowDeskBtn
Win7ShowDeskBtnAero

I think showing the tooltip (like Win7) is better than assuming that the user already know what the button does.

**Review by learn-more — APPROVED** [review]

File: base/shell/explorer/lang/en-US.rc [hunk 1]
```diff
@@ -215,4 +215,5 @@ BEGIN
     IDS_RESTORE_ALL "&Show Open Windows"
     IDS_TRAYWND_UNDO_CASCADE "&Undo Cascade"
     IDS_TRAYWND_UNDO_TILE "&Undo Tile"
+    IDS_TRAYDESKBTN_TOOLTIP "Show Desktop/Windows"
```
Comment (HBelusca) [inline base/shell/explorer/lang/en-US.rc:218]: If I were you I would avoid the capitalization here. 

```suggestion
    IDS_TRAYDESKBTN_TOOLTIP "Show desktop/windows"
```
Cc @DarkFire01 @cbialorucki @Doug-Lyons regarding "good practices" in English here.

**Review by HBelusca — APPROVED** [review]
Whatever the choice for the translation, lgtm.

**Comment by TAN-Gaming** [comment]
> Looks fine to me but we should perhaps use a tool like [MiniSpy](https://www.codeproject.com/articles/1732/minispy) to try to figure out if Windows has one tooltip window for everything on the taskbar or if each feature has its own tip HWND.

Checked on Win7 with WinSpy++. It uses separated tooltips for each feature.

File: base/shell/explorer/lang/en-US.rc (hunk 1 above)
Comment (TAN-Gaming) [reply base/shell/explorer/lang/en-US.rc:218]: Windows<7 and ReactOS uses uppercase but Windows7+ uses lowercase.

Win2k3:
VirtualBox_Win2003_31_10_2024_00_06_01

Win10:
Win10TaskbarCM

Win2k3 (Quick Launch):
VirtualBox_Win2003_31_10_2024_00_11_21

**Comment by RatinCN** [comment]
> > try to figure out if Windows has one tooltip
> 
> On Windows 10 I can't even get it to display a tooltip here (Peek on or off does not matter).

image

Seems there is  a tooltip on my Windows 11, the text is "Show Desktop" in English.

**Pull request merged**